app/src/pages/admin/serverSettingsPage/emailServerTab/emailServerTab.jsx

```diff
@@ -0,0 +1,195 @@
+import classNames from 'classnames/bind';
+import { SpinningPreloader } from 'components/preloaders/spinningPreloader';
+import { BigButton } from 'components/buttons/bigButton';
+import WarningIcon from 'common/img/error-inline.svg';
+import { EMAIL_SERVER_FORM, EMAIL_ENABLED_KEY, DEFAULT_FORM_CONFIG } from './constants';
+import { EmailServerTabFormFields } from './emailServerTabFormFields';
+import styles from './emailServerTab.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  emailSwitcher: {
+    id: 'EmailServerTab.emailSwitcher',
+    defaultMessage: 'Enable email server',
+  },
+  updateEmailServerSuccess: {
+    id: 'EmailServerTabFormFields.updateEmailServerSuccess',
+    defaultMessage: 'Email server settings have been updated',
+  },
+  warningMessage: {
+    id: 'EmailServerTabFormFields.warningMessage',
+    defaultMessage: 'Test connection was failed:',
+  },
+});
+
+@reduxForm({
+  form: 'emailServerTabForm',
+  validate: ({ host, port }) => ({
+    host: !host && 'requiredFieldHint',
+    port: (!port || !validate.inRangeValidate(port, 1, 65535)) && 'portFieldHint',
+  }),
+  initialValues: DEFAULT_FORM_CONFIG,
+})
+@connect(
+  (state) => ({
+    enabled: formValueSelector(EMAIL_SERVER_FORM)(state, [EMAIL_ENABLED_KEY]),
+  }),
+  {
+    showNotification,
+  },
+)
+@injectIntl
+export class EmailServerTab extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    enabled: PropTypes.bool,
+    initialize: PropTypes.func,
+    handleSubmit: PropTypes.func,
+    change: PropTypes.func,
+    showNotification: PropTypes.func,
+  };
+
+  static defaultProps = {
+    enabled: false,
+    initialize: () => {},
+    handleSubmit: () => {},
+    change: () => {},
+    showNotification: () => {},
+  };
+
+  state = {
+    loading: true,
+    errorMessage: '',
+  };
+
+  componentDidMount() {
+    this.fetchEmailConfig();
+  }
+
+  onFormSubmit = (data) => {
+    this.setState({
+      loading: true,
+      errorMessage: '',
+    });
+    const requestOptions = {
+      method: this.props.enabled ? 'POST' : 'DELETE',
+    };
+    if (this.props.enabled) {
+      requestOptions.data = data;
+    }
+    this.updateEmailConfig(requestOptions);
+  };
+
+  updateEmailConfig = (options) =>
+    fetch(URLS.emailServerSettings(), options)
+      .then(this.updateSettingSuccess)
+      .catch(this.catchRequestError);
+
+  fetchEmailConfig = () =>
+    fetch(URLS.serverSettings())
+      .then((data) => {
+        this.props.initialize({
+          [EMAIL_ENABLED_KEY]: true,
+          password: '',
+          username: '',
+          ...data.serverEmailConfig,
+        });
+        this.setState({
+          loading: false,
+        });
+      })
+      .catch(() => {
+        this.setState({
+          loading: false,
+        });
+      });
+
+  updateSettingSuccess = (response) => {
+    const {
+      intl: { formatMessage },
+    } = this.props;
+    if (response.msg.match('remove')) {
```

Comment:
Are there more stable ways to determine operation type?

Reply:
Done